### DISCUSSION TITLE
Odd issue when trying to open empty file with lf

### DISCUSSION THREAD

**Discussion by Samueru-sama** [Anomalocaridid/handlr-regex#47] [Q&A] [answered]
I have application/x-zerosize set to use mousepad's desktop entry.

If I am in thunar and click to open an empty file, it opens in mousepad without issues. 

however if I an in lf, which is a terminal based file manager, if I try to open an empty file I get an error: 

```
handlr error 
could not figure out the mime type of _'pathtofile'_
```

This only happens with empty files and it only happens with lf. If it is any other file type I have no issues opening it with its respective application. 

I'm not sure if this is an issue with lf or with handlr.

**Comment by Anomalocaridid** [comment]
How is `lf` calling `handlr`? I am not familiar with `lf` so please feel free to include any other information about it that you think is relevant or helpful.

Also what version of `handlr-regex` are you using?

**Reply by Samueru-sama** [reply]
> How is `lf` calling `handlr`? I am not familiar with `lf` so please feel free to include any other information about it that you think is relevant or helpful.
> 
> Also what version of `handlr-regex` are you using?

lf uses xdg-open, and I do have the xdg-open script that redirects to handlr open. 

handlr-regex is version v0.9.0

lf is easy to use to test the issue, just install and type lf in the terminal and navigate with the arrow keys to an empty file, then hit arrow right which will open the file.

**Reply by Anomalocaridid** [reply]
Sorry for forgetting about this for so long. Is this still an issue?

**Reply by Samueru-sama** [reply]
> Sorry for forgetting about this for so long. Is this still an issue?

Yes, here is a vid I just recorded of the issue: https://streamable.com/sjajdm

Thanks for keeping track and reminding of the issue, even I forgot lol.

**Reply by Anomalocaridid** [reply]
It seems to be a `handlr-regex` issue. I can replicate it even without using `lf`.

I think the issue's basically just that `handlr-regex` currently detects empty files with no file extension as `application/x-zerosize`, which results in the error message.

However, `xdg-mime query filetype` detects the same file as `text/plain`.

I have no clue how thunar works under the hood, but I assume it handles filetype detection internally, possibly using some library, rather than shelling out to `xdg-open`.

More specifically, I think it might be an issue with an upstream crate, `xdg-mime`. Fortunately I already had a PR open on it for a while that's intended to fix problems like this, https://github.com/ebassi/xdg-mime-rs/pull/30.

**Reply by Samueru-sama** [reply]
> It seems to be a `handlr-regex` issue. I can replicate it even without using `lf`.
> 
> I think the issue's basically just that `handlr-regex` currently detects empty files with no file extension as `application/x-zerosize`, which results in the error message.
> 
> However, `xdg-mime query filetype` detects the same file as `text/plain`.
> 
> I have no clue how thunar works under the hood, but I assume it handles filetype detection internally, possibly using some library, rather than shelling out to `xdg-open`.
> 
> More specifically, I think it might be an issue with an upstream crate, `xdg-mime`. Fortunately I already had a PR open on it for a while that's intended to fix problems like this, [ebassi/xdg-mime-rs#30](https://github.com/ebassi/xdg-mime-rs/pull/30).

But I already have my text editor (lite-xl) as the default for `application/x-zerosize` in the `$XDG_CONFIG_HOME/mimeapps.list` file? shouldn't that be enough?

**Reply by Anomalocaridid** [reply]
Currently, when `handlr` gets `application/x-zerosize` it just discards it, giving an error message.

Although it occurs to me that this behavior should probably be changed and `application/x-zerosize` should probably just be handled properly.

**Reply by Anomalocaridid** [reply, marked as answer]
Alright, this should be solved in v0.11.2.

**Reply by Samueru-sama** [reply]
@Anomalocaridid Thank you, it works!
